### PR TITLE
fix(signup): Stop polling after leaving the /confirm page.

### LINKED ISSUE #5323: `null has no properties` being reported in Sentry
https://sentry.prod.mozaws.net/operations/fxa-content-product/issues/632666/events/11503181/

I haven't tested this, but I believe the STRs are:

1. From the firstrun page, sign up to FxA for Sync.
2. Click through CWTS. 
3. On the `Verify your email` screen, click the browser's `back` button.
4. Verify the email.

There are two problems:

1. https://github.com/mozilla/fxa-content-server/blob/f9f7f10d4310163ffc256d1536322d470dcfe646/app/scripts/views/mixins/experiment-mixin.js#L61 doesn't check whether `experiments` can be dereferenced.
2. https://github.com/mozilla/fxa-content-server/blob/f9f7f10d4310163ffc256d1536322d470dcfe646/app/scripts/views/confirm.js#L111 starts a confirmation poll, but we never stop the poll when the view is destroyed.

Breaking the above into two separate issues.

### LINKED ISSUE #5325: confirmation poll is not stopped if the user clicks the browser's "back" button from /confirm
https://github.com/mozilla/fxa-content-server/blob/f9f7f10d4310163ffc256d1536322d470dcfe646/app/scripts/views/confirm.js#L111 

If the /confirm screen starts a confirmation poll and the user clicks the browser's "back" button, we never stop the poll, it just sits there polling in the background.

ref https://github.com/mozilla/fxa-content-server/issues/5323
ref https://github.com/mozilla/fxa-content-server/issues/4094

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Extract the logic to do the signup/signin verification polling
from models/account.js into its own model, models/polls/session.js

Extract most of the polling logic from /confirm into
its own mixin, session-verification-poll-mixin. This
will allow the common logic to be used on CWTS.

The poll is started when entering the /confirm page,
stopped when leaving.

The hope is that this will make #4094 and #4193 very easy.

fixes #5323
fixes #5325

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@mozilla/fxa-devs - r?

**Comment by philbooth** [comment]
LGTM, r+

**Comment by shane-tomlinson** [comment]
>  As a side note, the polling stops but if the user clicks forward, they will be an Invalid Token error.

I haven't been able to reproduce this particular bug. Can you list the STRs? 

Thanks to this tip though, I was able to find https://github.com/mozilla/fxa-content-server/issues/5347.

**Comment by shane-tomlinson** [comment]
> I haven't been able to reproduce this particular bug

@vbudhram - after enough noodling, I was able to reproduce this.

File: app/scripts/views/confirm.js [hunk 1]
```diff
@@ -87,11 +77,14 @@ define(function (require, exports, module) {
...
         )
-        .then(() => this._startPolling());
+        .then(() =>
+          this.waitForSessionVerification(account, () => this._gotoNextStep())
+        );
     },
 
-    _startPolling () {
-      return this._waitForConfirmation()
+    _gotoNextStep () {
```
Comment (shane-tomlinson) [inline app/scripts/views/confirm.js]: > Would something like _advancePollState be more clear?

It's not really a pollState either, it's the next screen. `_gotoNextScreen`?

Comment (vbudhram) [inline app/scripts/views/confirm.js]: That works for me!

**Pull request merged**